install.sh

```diff
@@ -1,62 +1,167 @@
+    sleep 1
+
+
+    # Check for known OS's
+    INSTALLED=0
+
+    case "$(uname -s)" in
+        Darwin*) install_macos && INSTALLED=1 || install_fail;;
+        # Mac
+        #CYGWIN*) do_something;;
+        # Cygwin Emulator -- Unimplemented
+        #MINGW*) do_something;;
+        # MINGW Emulator -- Unimplemented
+    esac
+
+    [[ -e /etc/debian_version ]] && install_debian && INSTALLED=1
+
+    [[ -e /etc/fedora-release ]] && install_fedora && INSTALLED=1
+
+    [[ -e  /etc/centos-release ]] && install_centos && INSTALLED=1
+
+    [[ -e /etc/arch-release ]] && install_arch && INSTALLED=1
+
+    [[ $INSTALLED -eq 0 ]] && printf "You are on an unknown system. You will have to install the required packages manually.\nContributions are welcome to add support for your system:\nhttps://github.com/ThioJoe/YT-Spammer-Purge" && exit 1
 
-TAG=$(curl https://api.github.com/repos/ThioJoe/YT-Spammer-Purge/releases/latest -s | jq .name -r)
-if  [[ $JQ -ne 0 ]]; then
-	if [[ -e /etc/debian_version ]]; then
-		sudo apt purge jq
-	elif [[ -e /etc/fedora-release ]]; then
-		sudo dnf remove jq
-	elif [[ -e /etc/centos-release ]]; then
-		sudo yum remove jq
-	elif [[ -e /etc/arch-release ]]; then
-		sudo pacman -Rs jq
-	fi
-else
-	echo "Did not uninstall jq as it was preinstalled before running this script."
+}
+
+install_latest_release () {
+    echo "We are now going to download the code for YT-Spammer-Purge."
+    sleep 1
+    git clone https://github.com/ThioJoe/YT-Spammer-Purge
+    cd YT-Spammer-Purge
+    git checkout -q -m $(git describe --abbrev=0 --tags)
+    # Go to latest release
+}
+
+install_MAIN () {
+    clear
+    echo "We will now install YT-Spammer-Purge. If this is not what you intend to do, exit within 5 seconds..."
+    sleep 5
+    echo "Installing."
+    # Check what OS we're running on
+
+    # Check if we already installed the requirements -- git is needed to check for which mode to run.
+    [[ $REQUIREMENTS_INSTALLED -eq 0 ]] && install_os_requirements
+
+    echo "--------------------------"
+
+    install_latest_release
+
+    # Since we've gotten python3 installed:
+
+    echo "--------------------------"
+    install_python_requirements
+
+    # Done!
+
+    printf "Dependencies and Program installed into .\YT-Spammer-Purge!\nNow follow these instructions to get a client_secrets.json file!\nhttps://github.com/ThioJoe/YT-Spammer-Purge/wiki/Instructions:-Obtaining-an-API-Key\n\nYou may run this script again inside your installation to update.\n"
+    exit 0
+}
+
+update () {
+    clear
+    echo "We will now update YT-Spammer-Purge. If this is not what you intend to do, exit within 5 seconds..."
+    sleep 5
+    echo "Continuing."
+    echo "Current version is $(git describe --abbrev=0 --tags)"
+    echo "Updating..."
+
+
+    git fetch origin
+    echo "Latest version is $(git describe origin --abbrev=0 --tags)"
+    echo "Updating to this version."
+    git checkout -q -m $(git describe origin --abbrev=0 --tags)
```

Comment:
Quote this to prevent word splitting.
```suggestion
    git checkout -q -m "$(git describe origin --abbrev=0 --tags)"
```

See for more details: https://github.com/koalaman/shellcheck/wiki/SC2046